File: EyeGuideiOS/EyeGuide/Models/Session.swift
import Foundation

struct Session: Codable, Identifiable, Sendable {
    let id: String
    let userId: String
    let startedAt: String
    let endedAt: String?
    let location: String?
    let sessionType: String

    enum CodingKeys: String, CodingKey {
        case id
        case userId = "user_id"
        case startedAt = "started_at"
        case endedAt = "ended_at"
        case location
        case sessionType = "session_type"
    }
}

struct SessionResponse: Codable, Sendable {
    let session: Session
}

struct SessionListResponse: Codable, Sendable {
    let sessions: [Session]
    let total: Int
    let limit: Int
    let offset: Int
}

File: EyeGuideAndroid/app/src/main/java/com/ZacharyZhang/eyeguide/data/model/Session.kt
package com.ZacharyZhang.eyeguide.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Session(
    val id: String,
    @SerialName("user_id") val userId: String,
    @SerialName("started_at") val startedAt: String,
    @SerialName("ended_at") val endedAt: String? = null,
    val location: String? = null,
    @SerialName("session_type") val sessionType: String,
)

@Serializable
data class CreateSessionRequest(
    @SerialName("session_type") val sessionType: String = "general",
    val location: String? = null,
)

@Serializable
data class SessionResponse(
    val session: Session? = null,
)

@Serializable
data class SessionListResponse(
    val sessions: List<Session>,
    val total: Int,
    val limit: Int,
    val offset: Int,
)
